pallets/elections/src/impls.rs

```diff
@@ -293,6 +299,45 @@ where
 
         T::ValidatorRewardsHandler::add_rewards(rewards);
     }
+
+    fn calculate_underperforming_validators() {
+        let thresholds = CurrentEraCommitteeKickOutThresholds::<T>::get();
+        let current_committee = T::SessionInfoProvider::current_committee();
+        for validator in current_committee {
+            let underperformance = match SessionValidatorBlockCount::<T>::try_get(&validator) {
+                Ok(block_count) => block_count <= thresholds.block_count_threshold,
+                Err(_) => true,
+            };
+            if underperformance {
+                Self::mark_validator_underperformance(&thresholds, &validator)
+            }
+        }
+    }
+
+    fn mark_validator_underperformance(
+        thresholds: &CommitteeKickOutThresholds,
+        validator: &T::AccountId,
+    ) {
+        UnderperformedValidatorSessionCount::<T>::mutate(&validator, |count| {
+            *count += 1;
+        });
+        let counter = UnderperformedValidatorSessionCount::<T>::get(&validator);
```

Comment:
`mutate` returns the result of the closure passed as its argument, so you can save it in the line above.

Reply:
good point, done